srcpkgs/syncthingtray/template

```diff
@@ -0,0 +1,79 @@
+# Template file for 'syncthingtray'
+pkgname=syncthingtray
+version=1.2.2
+revision=1
+build_style=cmake
+configure_args="-DNO_FILE_ITEM_ACTION_PLUGIN=$(vopt_if kde OFF ON)
+ -DNO_PLASMOID=$(vopt_if kde OFF ON) -DWEBVIEW_PROVIDER=$(vopt_if webengine webengine none)
+ -DJS_PROVIDER=qml -DBUILTIN_TRANSLATIONS=ON -DSYSTEMD_SUPPORT=OFF -DBUILD_SHARED_LIBS=ON"
+hostmakedepends="pkg-config qt5-host-tools qt5-qmake $(vopt_if kde 'extra-cmake-modules plasma-framework')"
+makedepends="cpp-utilities-devel qtutilities-devel qtforkawesome-devel
+ qt5-devel qt5-core qt5-network qt5-gui qt5-widgets qt5-svg qt5-declarative
+ qt5-quickcontrols qt5-location-devel boost-devel
+ $(vopt_if webengine 'qt5-webengine-devel qt5-webchannel-devel')
+ $(vopt_if kde 'kio-devel plasma-framework-devel')"
+depends="syncthing"
+checkdepends="syncthing iproute2"
+short_desc="Tray application for Syncthing"
+maintainer="classabbyamp <void@placeviolette.net>"
+license="GPL-2.0-or-later"
+homepage="https://github.com/Martchus/syncthingtray"
+distfiles="https://github.com/Martchus/syncthingtray/archive/refs/tags/v${version}.tar.gz"
+checksum=0f556e30091ffc06e7e489c5e09e5460401c28fd1e8318b68cb6066ead7fe7a1
+
+make_check_pre="env SYNCTHING_TEST_TIMEOUT_FACTOR=3"
+make_check_target="check"
+
+build_options="kde webengine"
+desc_option_kde="Build with KDE support"
+desc_option_webengine="Build with Qt5 WebEngine support"
+build_options_default="kde"
+
+# if not enabled, it will fall back to web browser
+if [ "$XBPS_TARGET_ENDIAN" = "le" ]; then
+	if [ "$XBPS_WORDSIZE" = "$XBPS_TARGET_WORDSIZE" ]; then
+		build_options_default+=" webengine"
+	fi
+fi
+
+if [ -n "$XBPS_CHECK_PKGS" ]; then
+	makedepends+=" libcppunit-devel"
+fi
+
+pre_check() {
+	export SYNCTHING_PORT=$(comm -23 <(seq 49152 65535) <(ss -tan | awk '{print $4}' | cut -d':' -f2 | grep "[0-9]\{1,5\}" | sort | uniq) | shuf | head -n 1)
```

Comment:
O_O

Is this even worth doing?

Reply:
I grabbed it from the AUR package (maintained by the author). it just finds a free port in that range for tests to use.

Reply:
Yeah, I'm just wondering if it's worth doing this dance to find a port, or if checks should just be disabled.

Reply:
it's a small number of tests, so I don't see any issue with keeping it